src/app/api/movies/[movieId]/route.ts

```diff
@@ -0,0 +1,9 @@
+import { db } from '@/utils/db';
+import Movie from '@/models/movie.model';
+import { NextResponse } from 'next/server';
+db();
+
+export async function GET(request: any, { params }: any) {
+  const movie = await Movie.findById(params.movieId);
+  return NextResponse.json({ movie });
```

Comment:
Here you're sending the object exactly as it comes out of the database. That's risky, because it means that if you ever add anything secret on the model (e.g. an internal rating, or a publish state, or something like that) it will leak here.

For that reason, I recommend using code that explicitly responds with only the fields that you want to include, e.g.: 

```ts
return NextResponse.json({
  id: movie._id,
  title: movie.title,
  // Whatever other fields you need to include
});
```